File: backend/src/main/java/com/smartcampus/backend/dto/search/history/SaveSearchHistoryRequestDTO.java
package com.smartcampus.backend.dto.search.history;

import com.smartcampus.backend.dto.search.ItineraryOptionDTO;
import com.smartcampus.backend.dto.search.SearchResultDTO;
import lombok.Getter;
import lombok.Setter;

//Envoye par le client quand il choisit de sauvegarder un itineraire parmi

@Getter
@Setter
public class SaveSearchHistoryRequestDTO {
    private SearchResultDTO origin;
    private SearchResultDTO destination;
    private ItineraryOptionDTO itinerary;
}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/search/SaveSearchHistoryRequestDto.kt
// model/search/SaveSearchHistoryRequestDto.kt
package com.ObservatoireCampus.mobile.model.search

/**
 * Miroir de SaveSearchHistoryRequestDTO.java cote backend.
 */
data class SaveSearchHistoryRequestDto(
    val origin: SearchResultDto,
    val destination: SearchResultDto,
    val itinerary: ItineraryOptionDto
)